src/server/storage_engine/default_engine.rs

```diff
@@ -100,30 +103,39 @@ impl StorageEngine for DefaultEngine {
         self.fsck().unwrap();
     }
 
-    fn get_file_attributes(&self, path: String) -> Result<Option<Vec<String>>, EngineError> {
+    fn get_file_attributes(&self, path: String) -> Result<Vec<u8>, EngineError> {
         match self.file_attr_db.db.get(path.as_bytes())? {
-            Some(value) => Ok(Some(bincode::deserialize(&value[..]).unwrap())),
-            None => Ok(None),
+            Some(value) => {
+                debug!("path: {}, value: {:?}", path, value);
+                Ok(value)
+            }
+            None => Err(EngineError::NoEntry),
         }
     }
 
-    fn read_directory(&self, path: String) -> Result<Vec<String>, EngineError> {
+    fn read_directory(&self, path: String) -> Result<Vec<u8>, EngineError> {
         match self.file_attr_db.db.get(path.as_bytes())? {
             Some(value) => {
-                if "d" != bincode::deserialize::<String>(&value[..]).unwrap() {
+                debug!("read_dir getting attr, path: {}, value: {:?}", path, value);
+                if b'd' != value[0] {
+                    debug!("deserialize err path: {}, value: {:?}", path, value);
                     return Err(EngineError::NotDir);
                 }
             }
             None => {}
         }
+        debug!("read_dir path: {}", path);
         match self.dir_db.db.get(path.as_bytes())? {
             Some(value) => {
                 let sub_dirs: SubDirectory = bincode::deserialize(&value[..]).unwrap();
-                let mut sub_dirs_str = Vec::new();
+                // let mut sub_dirs_str = Vec::new();
                 for sub_dir in sub_dirs.sub_dir {
-                    sub_dirs_str.push(sub_dir.0);
+                    //     sub_dirs_str.push(sub_dir.0);
+                    debug!("read_dir sub_dir: {}", sub_dir.0);
                 }
-                Ok(sub_dirs_str)
+                // Ok(sub_dirs_str)
+                debug!("read_dir path: {}, value: {:?}", path, value);
```

Comment:
It seems that these lines can be deleted.

Reply:
Debug and comment already simplified.